File: 10/JackAnalyzer.java
package comsys.compiler;

import java.io.File;
import java.util.Arrays;

public class JackAnalyzer {

    public static void main(String[] args) {
        String filePath = "Square\\";

        JackAnalyzer analyzer = new JackAnalyzer();
        analyzer.analyze(filePath);
    }

    JackTokenizer tokenizer;
    CompilationEngine engine;
    static String prePath = "C:\\Users\\PC\\Desktop\\ComSysStudy\\nand2tetris\\projects\\10\\";

    void analyze(String filePath) {
        final File file = new File(prePath + filePath);

        if(file.isDirectory()) {
            Arrays.stream(file.list()).filter(f -> f.matches(".*\\.jack")).forEach(f -> {
                codeGenerate(file.getAbsolutePath() + "\\" + f
                        , file.getAbsolutePath() + "\\out\\" + f.replace(".jack", ".xml"));
            });
        } else {
            codeGenerate(file.getAbsolutePath(), file.getParent() + "\\out\\" + file.getName().replace(".jack", ".xml"));
        }
    }

    void codeGenerate(String inputFilePath, String outputFilePath) {
        try {
            tokenizer = new JackTokenizer(inputFilePath);
            engine = new CompilationEngine(tokenizer, outputFilePath);

            tokenizer.tokenize();
            engine.compile();
        } catch (Exception e) {}
    }
}

File: 10/jackanalyzer.py
import sys, os, os.path, glob
import jacktokenizer
import compilationengine
class JackAnalyzer:
    def __init__(self):
        pass

    def check(self, inputfiles, outputfile):
        for inputfile in inputfiles:
            #jacktokenizer.tokenizer(inputfile)
            #tokenizer = jacktokenizer.JackTokenizer(inputfile)
            #print(tokenizer.tokens)
            compiler = compilationengine.CompilationEngine(inputfile, outputfile)


if sys.argv[1].endswith('.jack'): # 하나의 파일
    inputfiles = [sys.argv[1]]
    outputfile = sys.argv[1].replace('.jack','.xml') # 파일명으로 파일 생성
else: # 디렉터리명
    inputfiles = glob.glob(sys.argv[1]+'/*.jack')
    outputfile = sys.argv[1]+'/'+sys.argv[1]+'.xml' # 디렉터리명으로 파일 생성

analyzer = JackAnalyzer()
analyzer.check(inputfiles, outputfile)
